src/main/java/org/quantumbadger/redreader/activities/OptionsMenuUtility.java

```diff
@@ -82,185 +89,285 @@
-				if(optionsMenuItemsPrefs.contains(OptionsMenuItemsPref.SEARCH)) add(activity, menu, Option.SEARCH, false);
+				add(activity, menu, Option.SUBMIT_POST, getOrThrow(appbarItemsPrefs, AppbarItemsPref.SUBMIT_POST), false);
 
-				if(subredditPinned != null && optionsMenuItemsPrefs.contains(OptionsMenuItemsPref.PIN)) {
+				if(subredditPinned != null) {
 					if(subredditPinned) {
-						add(activity, menu, Option.UNPIN, false);
+						add(activity, menu, Option.UNPIN, getOrThrow(appbarItemsPrefs, AppbarItemsPref.PIN), false);
 					} else {
-						add(activity, menu, Option.PIN, false);
+						add(activity, menu, Option.PIN, getOrThrow(appbarItemsPrefs, AppbarItemsPref.PIN), false);
+					}
+				}
+
+				if(subredditBlocked != null) {
+					if(subredditBlocked) {
+						add(activity, menu, Option.UNBLOCK, getOrThrow(appbarItemsPrefs, AppbarItemsPref.BLOCK), false);
+					} else {
+						add(activity, menu, Option.BLOCK, getOrThrow(appbarItemsPrefs, AppbarItemsPref.BLOCK), false);
 					}
 				}
 
 				if(subredditSubscriptionState != null) {
-					addSubscriptionItem(activity, menu, subredditSubscriptionState);
+					addSubscriptionItem(activity, menu, getOrThrow(appbarItemsPrefs, AppbarItemsPref.SUBSCRIBE), subredditSubscriptionState);
 				}
 
-				if(subredditHasSidebar) add(activity, menu, Option.SIDEBAR, false);
+				if(subredditHasSidebar) {
+					add(activity, menu, Option.SIDEBAR, getOrThrow(appbarItemsPrefs, AppbarItemsPref.SIDEBAR), false);
+				}
 			}
-			if(commentsVisible) add(activity, refreshMenu, Option.REFRESH_COMMENTS, true);
 		}
 
-		if(optionsMenuItemsPrefs.contains(OptionsMenuItemsPref.ACCOUNTS)) add(activity, menu, Option.ACCOUNTS, false);
-		if(optionsMenuItemsPrefs.contains(OptionsMenuItemsPref.THEME)) add(activity, menu, Option.THEMES, false);
-		add(activity, menu, Option.SETTINGS, false);
-		if(optionsMenuItemsPrefs.contains(OptionsMenuItemsPref.CLOSE_ALL)) add(activity, menu, Option.CLOSE_ALL, false);
+		add(activity, menu, Option.ACCOUNTS, getOrThrow(appbarItemsPrefs, AppbarItemsPref.ACCOUNTS), false);
+		add(activity, menu, Option.THEMES, getOrThrow(appbarItemsPrefs, AppbarItemsPref.THEME), false);
+
+		//Always show settings if the main menu is visible, to prevent user from being locked out of them
+		if(subredditsVisible && getOrThrow(appbarItemsPrefs, AppbarItemsPref.SETTINGS) == DO_NOT_SHOW) {
+			add(activity, menu, Option.SETTINGS, MenuItem.SHOW_AS_ACTION_NEVER,false);
+		} else {
+			add(activity, menu, Option.SETTINGS, getOrThrow(appbarItemsPrefs, AppbarItemsPref.SETTINGS),false);
+		}
+
+		add(activity, menu, Option.CLOSE_ALL, getOrThrow(appbarItemsPrefs, AppbarItemsPref.CLOSE_ALL), false);
+
+		//Will be called from CommentListingFragment (after potentially adding the Reply button) if comments are shown
+		if(!commentsVisible) pruneMenu(activity, menu, appbarItemsPrefs, !subredditsVisible);
```

Comment:
It's probably worth calling it from here anyway, even if it's called a second time at some later point. To handle rotation changes, `pruneMenu()` will need to be callable multiple times, and "reset" everything to the correct state each time, and if so calling it twice here isn't an issue.